app/features/transactions/transaction-repository.ts

```diff
@@ -29,6 +40,51 @@ export class TransactionRepository {
     return TransactionRepository.toTransaction(data);
   }
 
+  async list({
+    userId,
+    cursor = null,
+    pageSize = 25,
+    abortSignal,
+  }: ListOptions) {
+    let query = this.db
+      .from('transactions')
+      .select()
+      .eq('user_id', userId)
+      .in('state', ['PENDING', 'COMPLETED'])
+      .order('created_at', { ascending: false })
+      .order('id', { ascending: false })
+      .limit(pageSize);
+
+    if (cursor) {
+      query = query.or(
+        `created_at.lt.${cursor.createdAt},and(created_at.eq.${cursor.createdAt},id.lt.${cursor.id})`,
```

Comment:
unfortunately I don't see how to do multi column comparison via supabase-js. postgres does have that feature and the query with that would be more performant as wouldn't need to have or in the where clause. I asked the question [here](https://github.com/orgs/supabase/discussions/35624) so we will see what they say

it's more nice to have though. it should be decent even without that